build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle

```diff
@@ -0,0 +1,66 @@
+import java.nio.charset.Charset
+import java.util.concurrent.TimeUnit
+
+/**
+ * This Gradle script captures the processor architecture
+ * and adds these as a custom value.
+ * This should be applied to the root project:
+ * <code> apply from: file('gradle-processor-arch.gradle') </code>
+ */
+
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+  return
+}
+
+buildScan.background { api ->
+  captureProcessorArch(api)
+}
+
+static void captureProcessorArch(def api) {
+  def osName = System.getProperty("os.name")
+  api.value("os.name", osName)
+
+  def osArch = System.getProperty("os.arch")
+  api.value("os.arch", osArch)
+
+  if (isDarwin(osName)) {
+    if (isM1()) {
+      api.tag("M1")
+    }
+    if (isTranslatedByRosetta()) {
+      api.tag("proc_translated")
+    }
+  }
+}
+
+static boolean isDarwin(String osName) {
+  return osName.contains("OS X") || osName.startsWith("Darwin")
+}
+
+static boolean isM1() {
+  return execAndGetStdout("uname", "-p") == "arm"
+}
+
+// On Apple silicon, a universal binary may run either natively or as a translated binary
+// https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment#Determine-Whether-Your-App-Is-Running-as-a-Translated-Binary
+static boolean isTranslatedByRosetta() {
+  return execAndGetStdout("sysctl", "sysctl.proc_translated") == "sysctl.proc_translated: 1"
+}
+
+static String execAndGetStdout(String... args) {
```

Comment:
Is this implementation how we read from an Exec command in other places?

Reply:
Yes, but those files are meant to be standalone given the header instructions:
`apply from: file('gradle-processor-arch.gradle')`

I'm ok factorizing but my opinion is that a sample should be self sufficient, wdyt?

Reply:
I think Etienne's comment refers to the fact that this is not consistent with the CCUD plugin's way of doing this.
It is, however, consistent with the other sample scripts so I propose keeping this PR as is and doing a follow up PR to make the samples consistent with CCUD.

Reply:
Yes, I was after consistency between scripts, not to share the code. All good it seems.